// shimmer-effects-react/dist/components/ShimmerText/ShimmerText.js
import { __makeTemplateObject } from "tslib";
import React from 'react';
import styled, { css } from 'styled-components';
import { shimmerAnimation } from '../ShimmerButton/ShimmerButton';
function gradientColors(props) {
    if (props.mode === "custom") {
        return [props.from, props.via, props.to];
    }
    else {
        return ["", "", ""];
    }
}
function getWidth(index, line, width) {
    if (line == 1) {
        return width;
    }
    if (index + 1 == line) {
        if (width != undefined) {
            return (width / 3);
        }
        else {
            return width;
        }
    }
    else {
        return width;
    }
}
export default function ShimmerText(props) {
    var _a = gradientColors(props), from = _a[0], via = _a[1], to = _a[2];
    return (React.createElement("div", { className: props.className, style: { display: "grid", placeItems: props.center ? "center" : "start" } },
        (props.loading && !props.children) &&
            Array.from({ length: props.line }).map(function (_, index) {
                return React.createElement(Text, { className: "shimmer-text-line-".concat(index), key: index, mode: props.mode, height: props.height, width: getWidth(index, props.line, props.width), border: props.border, rounded: props.rounded, line: props.line, gap: props.gap, from: from, via: via, to: to });
            }),
        (props.loading && props.children) ?
            Array.from({ length: props.line }).map(function (_, index) {
                return React.createElement(Text, { className: "shimmer-text-line-".concat(index), key: index, mode: props.mode, height: props.height, width: getWidth(index, props.line, props.width), border: props.border, rounded: props.rounded, line: props.line, gap: props.gap, from: from, via: via, to: to });
            })
            : props.children));
}
ShimmerText.defaultProps = {
    height: 8,
    width: 100,
    border: 1,
    rounded: 1,
    gap: 6,
    loading: true,
    center: false,
    className: 'shimmer-text-div'
};
var Text = styled.div(templateObject_8 || (templateObject_8 = __makeTemplateObject(["\n\n    ", "\n\n    ", "  \n\n    ", " \n\n    ", "\n\n    ", "\n\n    ", "\n\n    ", ";\n    \n    background-size: 1000px 100%;\n    \n    animation: ", " 4.2s linear infinite forwards;\n\n"], ["\n\n    ", "\n\n    ", "  \n\n    ", " \n\n    ", "\n\n    ", "\n\n    ", "\n\n    ", ";\n    \n    background-size: 1000px 100%;\n    \n    animation: ", " 4.2s linear infinite forwards;\n\n"])), function (p) { return p.mode == "light" && css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n        background: linear-gradient(to right, #cbd5e1 8%, #f1f5f9 18%, #cbd5e1 33%);\n     "], ["\n        background: linear-gradient(to right, #cbd5e1 8%, #f1f5f9 18%, #cbd5e1 33%);\n     "]))); }, function (p) { return p.mode == "dark" && css(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n        background: linear-gradient(to right, #374151 8%, #6b7280 18%, #374151 33%);\n     "], ["\n        background: linear-gradient(to right, #374151 8%, #6b7280 18%, #374151 33%);\n     "]))); }, function (p) { return p.mode == "custom" && css(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n        background: linear-gradient(to right, ", " 8%, ", "  18%, ", "  33%);\n     "], ["\n        background: linear-gradient(to right, ", " 8%, ", "  18%, ", "  33%);\n     "])), p.from, p.via, p.to); }, function (p) { return (p.height && p.width) && css(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n      height: ", "px;\n      width: ", "%;\n     "], ["\n      height: ", "px;\n      width: ", "%;\n     "])), p.height, p.width); }, function (p) { return p.border && css(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n     border-width: ", "px;\n     "], ["\n     border-width: ", "px;\n     "])), p.border); }, function (p) { return p.rounded && css(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n     border-radius: ", "em;\n     "], ["\n     border-radius: ", "em;\n     "])), p.rounded); }, function (p) { return p.gap && css(templateObject_7 || (templateObject_7 = __makeTemplateObject(["\n     margin-bottom: ", "px;\n     "], ["\n     margin-bottom: ", "px;\n     "])), p.gap); }, shimmerAnimation);
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7, templateObject_8;
