// shimmer-effects-react/dist/components/ShimmerTable/ShimmerTable.js
import { __makeTemplateObject } from "tslib";
import React from "react";
import ShimmerText from "../ShimmerText/ShimmerText";
import styled, { css } from "styled-components";
function gradientColors(props) {
    if (props.mode === "custom") {
        return [props.from, props.via, props.to];
    }
    else {
        return ["", "", ""];
    }
}
export default function ShimmerTable(props) {
    var _a = gradientColors(props), from = _a[0], via = _a[1], to = _a[2];
    return (React.createElement("div", { className: "shimmer-table-main-div" },
        props.loading && !props.children && (React.createElement(Table, { className: "shimmer-table", rowGap: props.rowGap }, Array.from({ length: props.row }).map(function (_, rindex) {
            return (React.createElement(TableRows, { className: "shimmer-table-row-".concat(rindex), key: rindex }, Array.from({ length: props.col }).map(function (_, cindex) {
                return (React.createElement(TableCols, { className: "shimmer-table-row-".concat(rindex, "-col-").concat(cindex), key: cindex, mode: props.mode, border: props.border, borderColor: props.borderColor, rounded: props.rounded, colPadding: props.colPadding, from: from },
                    React.createElement(ShimmerText, { className: "shimmer-table-row-".concat(rindex, "-col-").concat(cindex, "-text-div"), mode: props.mode, line: 1, gap: 0, width: 100, from: from, via: via, to: to })));
            })));
        }))),
        props.loading && props.children ? (React.createElement(Table, { className: "shimmer-table", rowGap: props.rowGap }, Array.from({ length: props.row }).map(function (_, rindex) {
            return (React.createElement(TableRows, { className: "shimmer-table-row-".concat(rindex), key: rindex }, Array.from({ length: props.col }).map(function (_, cindex) {
                return (React.createElement(TableCols, { className: "shimmer-table-row-".concat(rindex, "-col-").concat(cindex), key: cindex, mode: props.mode, border: props.border, borderColor: props.borderColor, rounded: props.rounded, colPadding: props.colPadding, from: from },
                    React.createElement(ShimmerText, { className: "shimmer-table-row-".concat(rindex, "-col-").concat(cindex, "-text-div"), mode: props.mode, line: 1, gap: 0, width: 100, from: from, via: via, to: to })));
            })));
        }))) : (props.children)));
}
;
ShimmerTable.defaultProps = {
    border: 1,
    rounded: 0.1,
    loading: true,
    rowGap: 14,
    colPadding: [10, 5, 10, 5],
    className: "shimmer-table"
};
var Table = styled.table(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n  border-collapse: separate;\n  width: 100%;\n\n  ", "\n"], ["\n  border-collapse: separate;\n  width: 100%;\n\n  ", "\n"])), function (p) {
    return css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n      border-spacing: 0 ", "px;\n    "], ["\n      border-spacing: 0 ", "px;\n    "])), p.rowGap);
});
var TableRows = styled.tr(templateObject_3 || (templateObject_3 = __makeTemplateObject([""], [""])));
var TableCols = styled.td(templateObject_8 || (templateObject_8 = __makeTemplateObject(["\n  margin-left: auto;\n  margin-right: auto;\n\n  ", "\n\n  ", "\n  \n      ", "\n  \n      ", "\n"], ["\n  margin-left: auto;\n  margin-right: auto;\n\n  ", "\n\n  ", "\n  \n      ", "\n  \n      ", "\n"])), function (p) {
    return p.mode == "light" && css(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n      border: ", "px solid ", ";\n      border-radius: ", "em;\n    "], ["\n      border: ", "px solid ", ";\n      border-radius: ", "em;\n    "])), p.border, p.borderColor, p.rounded);
}, function (p) {
    return p.mode == "dark" && css(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n      border: ", "px solid ", ";\n      border-radius: ", "em;\n    "], ["\n      border: ", "px solid ", ";\n      border-radius: ", "em;\n    "])), p.border, p.borderColor, p.rounded);
}, function (p) {
    return p.mode == "custom" && css(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n      border: ", "px solid ", ";\n      border-radius: ", "em;\n    "], ["\n      border: ", "px solid ", ";\n      border-radius: ", "em;\n    "])), p.border, p.from, p.rounded);
}, function (p) {
    return css(templateObject_7 || (templateObject_7 = __makeTemplateObject(["\n      padding: ", "px ", "px ", "px\n        ", "px;\n    "], ["\n      padding: ", "px ", "px ", "px\n        ", "px;\n    "])), p.colPadding[0], p.colPadding[1], p.colPadding[2], p.colPadding[3]);
});
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7, templateObject_8;
